tests/tracer/test_span.py

```diff
@@ -479,6 +479,12 @@ def test_span_unicode_set_tag():
     span._set_str_tag(u"😐", u"😌")
 
 
+def test_span_bad_unicode_set_tag():
+    span = Span(None, None)
+    span.set_tag("key", "🤔")
+    span._set_str_tag("key", "🤔")
```

Comment:
I think we should be able to handle this case in Python 2. Right now `stringify` is just `unicode` in Python 2 which defaults to interpreting the input string as ascii encoded. If we did `unicode(s, "utf-8")` we'd still be able to encode strings like this.

➕ for still doing the error handling as the data might not necessarily be utf-8 or ascii (as seen in the flask case recently).

Reply:
Does that mean that we'll push invalid data without having a way for the users to be noticed unless they dig into their logs?

Reply:
Right @jd this would require someone to notice it in their logs. I don't think we have yet a better behavior to offer yet since this would be an error that we'd cause while tracing rather than at startup.

Reply:
The way I understand it is that they are 2 cases:

1. The user triggers this problem: they did a programming error and need to fix it because they're sending the wrong data type to ddtrace. 
2. We have a bug in an integration that needs to be fixed.

You have two choices:
a. Ignore (catch the exception and ignore/log it)
b. Fix it

This is picking `a` for 1 and 2 IIUC.

For 1. I don't think choosing for the user is a good experience. Once you pick `a. Ignore` there is no way for them to decide anything else. As a user, I dislike libraries hiding my bugs.

For 2. As a ddtrace maintainer, I'm not sure hiding integration issues in user's log is going to help us improve the user experience and quality of our integration

Thoughts?